sihl/src/sihl.mli

```diff
@@ -315,6 +486,91 @@ module Web : sig
   end
 end
 
+module Log : sig
+  include module type of Core_log
+end
+
+module Cleaner : sig
+  include module type of Core_cleaner
+end
+
+module Container : sig
+  (** A module to manage the service container and service lifecycles.
+
+      The service container knows how to start services in the right order by
+      respecting the defined dependencies. Use it to implement your own
+      services. *)
+
+  (** {1 Lifecycle}
+
+      Every service has a lifecycle, meaning it can be started and stopped. **)
+
+  type lifecycle = Core_container.lifecycle =
+    { name : string
+    ; dependencies : unit -> lifecycle list
+    ; start : unit -> unit Lwt.t
+    ; stop : unit -> unit Lwt.t
+    }
+
+  val create_lifecycle
+    :  ?dependencies:(unit -> lifecycle list)
+    -> ?start:(unit -> unit Lwt.t)
+    -> ?stop:(unit -> unit Lwt.t)
+    -> string
+    -> lifecycle
+
+  (** {1 Service}
+
+      A service has a [start] and [stop] functions and a lifecycle. **)
+
+  module Service : sig
+    module type Sig = sig
+      val lifecycle : lifecycle
+    end
+
+    type t = Core_container.Service.t =
+      { lifecycle : lifecycle
+      ; configuration : Configuration.t
+      ; commands : Command.t list
+      ; server : bool
+      }
+
+    val commands : t -> Command.t list
+    val configuration : t -> Configuration.t
+
+    val create
+      :  ?commands:Command.t list
+      -> ?configuration:Configuration.t
+      -> ?server:bool
+      -> lifecycle
+      -> t
+
+    val server : t -> bool
+    val start : t -> unit Lwt.t
+    val stop : t -> unit Lwt.t
+    val name : t -> string
+  end
+
+  (** [start_services lifecycles] starts a list of service [lifecycles]. The
+      order does not matter as the services are started in the order of their
+      dependencies. (No service is started before its dependency) *)
+  val start_services : Service.t list -> lifecycle list Lwt.t
+
+  (** [stop_services ctx services] stops a list of service [lifecycles] with a
```

Comment:
stops a list of `services` with a

Reply:
Fixed.

Reply:
Sorry, my correction was not clear. I meant that you mention `lifecycles` even though there is no argument like that.

Reply:
Fixed.

Reply:
Sorry, am I missing something? It still says
```
  (** [stop_services ctx services] stops a list of services [lifecycles] with a
```
In my diff?

Reply:
Sorry I fixed `start_services` which was also wrong, should be fixed now.

Reply:
It still says:
```
(** [stop_services ctx services] stops a list of [lifecycles] with a context
```
Did you mean to put `lifecycles` as an argument?

Reply:
@aronerben Should be fixed now. :)

Reply:
Cool, thanks :)